br/pkg/restore/snap_client/import.go

```diff
@@ -565,29 +551,34 @@ func (importer *SnapFileImporter) buildDownloadRequest(
 			RequestSource: kvutil.BuildRequestSource(true, kv.InternalTxnBR, kvutil.ExplicitTypeBR),
 		},
 	}
-	return req, *sstMeta, nil
+	return req, *sstMeta, false, nil
 }
 
 func (importer *SnapFileImporter) downloadSST(
 	ctx context.Context,
 	regionInfo *split.RegionInfo,
-	files []*backuppb.File,
-	rewriteRules *restoreutils.RewriteRules,
+	filesGroup []TableIDWithFiles,
 	cipher *backuppb.CipherInfo,
 	apiVersion kvrpcpb.APIVersion,
 ) ([]*import_sstpb.SSTMeta, error) {
 	var mu sync.Mutex
 	downloadMetasMap := make(map[string]import_sstpb.SSTMeta)
 	resultMetasMap := make(map[string]*import_sstpb.SSTMeta)
 	downloadReqsMap := make(map[string]*import_sstpb.DownloadRequest)
-	for _, file := range files {
-		req, sstMeta, err := importer.buildDownloadRequest(file, rewriteRules, regionInfo, cipher)
-		if err != nil {
-			return nil, errors.Trace(err)
+	for _, files := range filesGroup {
+		for _, file := range files.Files {
+			req, sstMeta, skip, err := importer.buildDownloadRequest(file, files.RewriteRules, regionInfo, cipher)
```

Comment:
It's better to use `req !=nil` to get rid of `skip` return  here.

Reply:
done